workers/unity/Assets/Playground/Scripts/Player/LocalPlayerInputSync.cs

```diff
@@ -16,17 +17,36 @@ public struct PlayerInputData
             public ComponentDataArray<Authoritative<SpatialOSPlayerInput>> PlayerInputAuthority;
         }
 
+        public VirtualJoystick VirtualJoystick;
+
         [Inject] private PlayerInputData playerInputData;
 
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+            GameObject controllerJoystick = GameObject.FindGameObjectWithTag("GameController");
+            VirtualJoystick = controllerJoystick.GetComponent<VirtualJoystick>();
+#if !(UNITY_ANDROID || UNITY_IOS)
+            controllerJoystick.setActive(false);
+#endif
+        }
+
         protected override void OnUpdate()
         {
             for (var i = 0; i < playerInputData.Length; i++)
             {
                 var cameraTransform = playerInputData.CameraTransform[i];
                 var forward = cameraTransform.Rotation * Vector3.up;
                 var right = cameraTransform.Rotation * Vector3.right;
+#if UNITY_ANDROID || UNITY_IOS
+                var input = Vector3.zero;
+                if (VirtualJoystick.InputDirection != Vector3.zero)
```

Comment:
no need for the if clause. if the input direction equals a zero vector the input variable will be zero as well.